relay/client/client.go

```diff
@@ -258,18 +258,22 @@ func (c *Client) handShake() error {
 		return err
 	}
 
-	msgType, err := messages.DetermineServerMsgType(buf[:n])
+	version, msgType, err := messages.DetermineMessageType(buf[:n])
 	if err != nil {
 		log.Errorf("failed to determine message type: %s", err)
 		return err
 	}
 
+	if version != messages.CurrentProtocolVersion {
```

Comment:
If on the server side we update the version then it won't be backward compatible. For example, if we extend the protocol with new message types that are useful for the new client, the old clients could also work because, in the current implementation, we drop the unknown message types.

Reply:
I think makes no sense to send the version info by the server. If the server knows about a client that it is not compatible then just close the communication by the server.

Reply:
>If on the server side we update the version then it won't be backward compatible. For example, if we extend the protocol with new message types that are useful for the new client, the old clients could also work because, in the current implementation, we drop the unknown message types.

We don't need to update the proto version if new message types are still compatible

>I think makes no sense to send the version info by the server. If the server knows about a client that it is not compatible then just close the communication by the server.

That's the behavior with only one version (!= expected version -> return error/close). If you have multiple the client can choose to keep supporting the older server version. Same thing on the server side